sentry-android-core/src/main/java/io/sentry/android/core/internal/tombstone/TombstoneParser.java

```diff
@@ -0,0 +1,219 @@
+
+    if (tombstone.hasSignalInfo()) {
+      final TombstoneProtos.Signal signalInfo = tombstone.getSignalInfo();
+      exception.setType(signalInfo.getName());
+      exception.setValue(excTypeValueMap.get(signalInfo.getName()));
+      exception.setMechanism(createMechanismFromSignalInfo(signalInfo));
+    }
+
+    exception.setThreadId((long) tombstone.getTid());
+    final List<SentryException> exceptions = new ArrayList<>(1);
+    exceptions.add(exception);
+
+    return exceptions;
+  }
+
+  @NonNull
+  private static Mechanism createMechanismFromSignalInfo(
+      @NonNull final TombstoneProtos.Signal signalInfo) {
+
+    final Mechanism mechanism = new Mechanism();
+    // this follows the current processing triggers strictly, changing any of these
+    // alters grouping and name (long-term we might want to have a tombstone mechanism)
+    mechanism.setType("signalhandler");
+    mechanism.setHandled(false);
+    mechanism.setSynthetic(true);
+
+    final Map<String, Object> meta = new HashMap<>();
+    meta.put("number", signalInfo.getNumber());
+    meta.put("name", signalInfo.getName());
+    meta.put("code", signalInfo.getCode());
+    meta.put("code_name", signalInfo.getCodeName());
+    mechanism.setMeta(meta);
+
+    return mechanism;
+  }
+
+  @NonNull
+  private Message constructMessage(@NonNull final TombstoneProtos.Tombstone tombstone) {
+    final Message message = new Message();
+    final TombstoneProtos.Signal signalInfo = tombstone.getSignalInfo();
+
+    // reproduce the message `debuggerd` would use to dump the stack trace in logcat
+    String command = String.join(" ", tombstone.getCommandLineList());
+    if (tombstone.hasSignalInfo()) {
+      String abortMessage = tombstone.getAbortMessage();
+      message.setFormatted(
+          String.format(
+              Locale.ROOT,
+              "%sFatal signal %s (%d), %s (%d), pid = %d (%s)",
+              !abortMessage.isEmpty() ? abortMessage + ": " : "",
+              signalInfo.getName(),
+              signalInfo.getNumber(),
+              signalInfo.getCodeName(),
+              signalInfo.getCode(),
+              tombstone.getPid(),
+              command));
+    } else {
+      message.setFormatted(
+          String.format(Locale.ROOT, "Fatal exit pid = %d (%s)", tombstone.getPid(), command));
+    }
+
+    return message;
+  }
+
+  private DebugMeta createDebugMeta(@NonNull final TombstoneProtos.Tombstone tombstone) {
+    final List<DebugImage> images = new ArrayList<>();
+
+    for (TombstoneProtos.MemoryMapping module : tombstone.getMemoryMappingsList()) {
+      // exclude anonymous and non-executable maps
+      if (module.getBuildId().isEmpty()
+          || module.getMappingName().isEmpty()
+          || !module.getExecute()) {
+        continue;
+      }
+      final DebugImage image = new DebugImage();
+      image.setCodeId(module.getBuildId());
+      image.setCodeFile(module.getMappingName());
+      image.setDebugId(module.getBuildId());
```

Comment:
I'm reading this [section](https://develop.sentry.dev/sdk/data-model/event-payloads/debugmeta/#elf-images:~:text=stack%20traces.-,debug_id,-Required.%20Debug%20identifier) and it says:

> the debug identifier is the little-endian UUID
representation of the first 16-bytes of that identifier

should we compute it here or symbolicator actually does that for us? And if it does it for us, should we leave it empty?